raven/utils.py

```diff
@@ -784,3 +777,86 @@ def generic_vector_reproject(
 
                 sink.write("{}".format(json.dumps(output)))
     return
+
+
+def zonalstats_raster_file(
+    stats: dict,
+    working_dir: str = None,
+    identifier: str = None,
+    raster_compression: str = RASTERIO_TIFF_COMPRESSION,
+    data_type: str = None,
+    crs: str = None,
+) -> str:
+    """
+    Extract the zonalstats grid(s) to a zipped GeoTIFF file and ensure that it is projected to the proper CRS.
+
+    Parameters
+    ----------
+    stats : dict
+      The dictionary produced by the rasterstats `zonalstats` function.
+    working_dir : str
+      The working directory (self.workdir).
+    identifier : str
+      The identifier of the process (self.identifier).
```

Comment:
Do you really need `identifier` or could it be a random name ?

Reply:
Remove reference to self.

Reply:
It could be a random name, certainly. Would it makes sense to just have a function make a random string and remove this field ?

Reply:
I think so.